src/engine/database/neo4j.rs

```diff
@@ -465,24 +465,31 @@ impl Transaction for Neo4jTransaction<'_> {
         }
 
         if !props.is_empty() {
+            let mut value_props : HashMap<String, Value> = HashMap::new();
             props.keys().enumerate().for_each(|(i, k)| {
```

Comment:
You may be able to work off the TODO below about taking ownership by replacing this with something like

```
props.into_iter().enumerate().for_each(|(i, (k, v))| { ... })
```